### ISSUE TITLE
keepalived: VIP gets inacccessible after a network issue

### ISSUE THREAD

**Issue by akop** [cloudfoundry-incubator/haproxy-boshrelease#167]
Hi,

we are using HAProxy bosh release 9.8.0 with keepalived job.

After a network issue the VIP got inaccessible.

The VIP moves to a backup node and after the network issue it releases it.
But the master doesn't know that and thinks he has the VIP still in use.

To solve this issue we have to restart keepalived job on master.

This topic seems to describe the same issue: https://serverfault.com/questions/821809/keepalived-send-gratuitous-arp-periodically

So we need to set the property vrrp_garp_master_refresh for keepalived.
Then the master will check periodically if it has the VIP, the default value is never.

I checked keepalived changelog and in Release 1.2.23 there is a fix: Fix unused global vrrp_garp_master_refresh.
So the 1.2.24 used in HAProxy bosh release should respect this option.

Can you please add this property to the keepalived job or add a common property where you can set all the keepalived options.

Also na update of the used keepalived version would be a good thing.

Thank you very much.

**Comment by domdom82** [comment]
Related to #217

**Comment by domdom82** [comment]
hi @akop we just bumped keepalived to v2.2.2 - can you confirm the issue still exists in this version of keepalived? if so, we can provide a PR to fix it.

**Comment by akop** [comment]
Hi @domdom82 I think since half of a year or even longer the issue didn't occurred again and I have no idea how to reproduce it. I guess it was something with OpenStack and DHCP. The stemcell had the same machine id for every VM so OpenStack DHCP was not extending the DHCP lease because it got asked from multiple VMs with the same machine id. 
So after 10min the host was losing the IP and then it was getting the same IP back from DHCP. This caused to break all active connections for example the Postgresql from apps running on diego-cell to Postgresql service. I guess keepalived had also to fight with that and this was the root cause of the problem.

At the end we disabled DHCP for BOSH managed VMs. I guess since then keepalived is also fine.

This machine-id not unique thing should be fixed already in newer stemcells because we were opening a issue for that and it's already fixed: https://github.com/cloudfoundry/bosh-linux-stemcell-builder/issues/122 

But thanks for this input I already saw it in the haproxy Slack channel and wanted to check if the new keepalived is supporting IPv6. Because this is a new requirement for us.

Not sure if we should keep this issue open to add a ability to add more special keepalived properties if they are needed.
